### PR TITLE
53035 http proxy

### PR DESCRIPTION
**Pull request by ricellis**
*What*
Add proxy support

*How*
Add `ConnectOptions` for proxy url, user and password.
Add a proxy auth interceptor.
Open connections using a `java.net.Proxy` when a proxy has been set.

*Testing*
Added a new test case that checks requests are sent to the proxy address and that the auth header is ok.

reviewer @rhyshort

### CODE REVIEW

File: src/main/java/com/cloudant/http/HttpConnection.java [hunk 1]
```diff
@@ -332,4 +334,28 @@ public void disconnect() {
...
+        if ("http".equals(proxyAddress.getProtocol()) || "https".equals(proxyAddress.getProtocol
+                ())) {
+            this.proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyAddress.getHost(),
+                    proxyAddress.getPort()));
+        } else {
+            throw new IllegalArgumentException("Only HTTP type proxies are supported");
+        }
+    }
+
+    public static String getEncodedAuth(String userInfo) throws IOException {
```
Comment (rhyshort) [inline src/main/java/com/cloudant/http/HttpConnection.java]: Delete this method

**Comment by rhyshort** [comment]
+1 after that minor comment is addressed

File: src/main/java/com/cloudant/http/HttpConnection.java (hunk 1 above)
Comment (ricellis) [inline src/main/java/com/cloudant/http/HttpConnection.java]: Fixed in e511f35

**Comment by emlaver** [comment]
+1

**Pull request merged**